src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -64,4 +70,21 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig)(implicit v
       }
     }
   }
+
+
+  def deleteCluster(googleProject: String, clusterName: String)(implicit executionContext: ExecutionContext): Future[Unit] = {
+    Future {
+      //currently, the bucketPath of the clusterRequest are not used - it will be used later as a place to store notebooks and results
+      val dataproc = getDataProc("dataproc")
+      val request = dataproc.projects().regions().clusters().delete(googleProject, dataprocConfig.dataprocDefaultZone, clusterName)
+      try {
+        executeGoogleRequest(request)
+      } catch {
+        case e:GoogleJsonResponseException =>
+          if(e.getStatusCode!=404)
```

Comment:
Why the `if(e.getStatusCode!=404)` check? Shouldn't we be catching _all_ `GoogleJsonResponseException`s and rethrowing `CallToGoogleApiFailedException`s?

Also there might be a subtle bug here. If a `GoogleJsonResponseException` is thrown with an status other than 404, the exception is caught, ignored, and the Future is successful! Which is probably not what we want. Whereas if we did this:
```
} catch {
  case e: GoogleJsonResponseException if e.getStatusCode!=404 =>
    throw CallToGoogleApiFailedException(googleProject, clusterName)
}
```
then the `catch` partial function would not match and the Future would be a failure with a `GoogleJsonResponseException` inside of it.

Reply:
If it's a 404 then the cluster is already gone and we don't have anything more to do.

Reply:
Or _do_ we want to legitimately ignore Google 404s? I'm confused. A comment would help. Also a log message -- I really don't silently catching and eating exceptions.

Reply:
Oh, duh. Well ok this is fine, I would still like a comment and a log message "cluster x is already deleted"